EIPS/eip-2569.md

```diff
@@ -0,0 +1,357 @@
+---
+eip: 2569
+title: Saving and Displaying Image Onchain for Universal Tokens
+author: Hua Zhang (@dgczhh), Yuefei Tan (@whtyfhas), Derek Zhou (@zhous)
+discussions-to: https://ethereum-magicians.org/t/erc-2569-saving-and-displaying-image-onchain-for-universal-tokens/4167
+status: Draft
+type: Standards Track
+category: ERC
+created: 2020-03-28
+---
+
+## Simple Summary
+A set of interfaces to save an SVG image in Ethereum, and to retrieve the image file from Ethereum for universal tokens. 
+
+## Abstract
+This set of interfaces allow a smart contract to save an SVG image in Ethereum and to retrieve an SVG image from Ethereum for fungible tokens, non-fungible tokens and tokens based on standards that will be developed in the future. 
+
+The interface set has two interfaces: one to save an SVG file in Ethereum and the other to retrieve an SVG file from Ethereum. 
+
+Typical applications include but not limited to:
+A solution for storage of a fungible token's icon.
+A solution for storage of a non-fungible token's icon.
+A solution for storage of the icon/logo of a DAO's reputation token.
+
+## Motivation
+The ERC-721 token standard is a popular standard to define a non-fungible token in Ethereum. This standard is widely used to specify a crypto gift, crypto medal, crypto collectible etc. The most famous use case is the [cryptokitty](https://www.cryptokitties.co/). 
+
+In most of these applications an image is attached to an ERC-721 token. For example, in the cryptokitty case each kitty has a unique image. While the token's code is saved in Ethereum permanently, the image attached to the token is not. 
+
+The existing solutions still keep such an image in a centralized server instead of Ethereum. When these applications display an image for a token they retrieve the token's information from Ethereum and search the centralized server for the token's associated image by using the token's information.
+
+Although this is an applicable way to display an image for a token, the image is still vulnerable to risks of being damaged or lost when saved in a centralized server.
+
+Hence we propose a set of interfaces to save an image for a universal token in Ethereum to keep the image permanent and tamper-resistant, and to retrieve an image for a universal token from Ethereum. 
+
+## Definitions
+```tokenId```: for a non-fungible token such as an ERC-721 token or a multi-token such as an ERC-1155 token which has a member "ID" to specify its token type or token index our proposed interface assigns an SVG image's file content to a string variable of the token's contract and  associates the SVG image to this "ID" number. This unique ID is used to access its SVG image in both a "set" operation and a "get" operation. 
```

Comment:
Use single backticks around inline code:
```
`tokenId`: for a non-fungible...
```

Reply:
Same issue in a number of places.

Reply:
Changes made and committed